hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -284,7 +284,10 @@ protected void commit(HoodieTable table, String commitActionType, String instant
     }
     // update Metadata table
     writeTableMetadata(table, instantTime, metadata);
-    activeTimeline.saveAsComplete(false, table.getMetaClient().createNewInstant(HoodieInstant.State.INFLIGHT, commitActionType, instantTime), Option.of(metadata));
+    HoodieInstant completedWriteInstant = activeTimeline.saveAsComplete(false,
```

Comment:
Should the `getTableFormat().commit(..)` work be pushed inside `activeTimeline.saveAsComplete(..)` .. So from this layer, it just feels like we are committing to the timeline per usual.. and internally it used native or a plugged-in table format